civis/service_client.py

```diff
@@ -0,0 +1,143 @@
+import civis
+from collections import OrderedDict
+import re
+import six
+import warnings
+
+from jsonref import JsonRef
+import requests
+
+from civis.base import Endpoint, CivisAPIError, CivisAPIKeyError, tostr_urljoin
+from civis.resources._resources import parse_method
+from civis._utils import to_camelcase
+
+
+def auth_service_session(session, service_id):
+    try:
+        service = civis.APIClient().services.get(service_id)
+    except CivisAPIError as err:
+        if err.status_code == 404:
+                    msg = 'There was an issue connecting to your service!'
+                    six.raise_from(ValueError(msg), err)
+        else:
+            raise
+
+    auth_url = service['current_deployment']['displayUrl']
+    # Make request for adding Authentication Cookie to session
+    session.get(auth_url)
+
+
+class ServiceEndpoint(Endpoint):
+
+    def __init__(self, session_kwargs, client,
+                 return_type='civis', root_path=None):
+        self._session_kwargs = session_kwargs
+        self._return_type = return_type
+        self._client = client
+        self._root_path = root_path
+
+    def _build_path(self, path):
+        if not path:
+            return self._client._base_url
```

Comment:
I changed it to be an attribute of ServiceClient, instead of (Service)Endpoint to try and reduce calls to the API. If it is an attribute of ServiceClient, then ServiceClient can use it when getting the swagger, and Endpoint can use it when making request calls. Instead of both classes needed to use the API to get the base path.